File: backend/school-entry/src/main/java/de/eshg/schoolentry/domain/model/ProcedureLabel.java
/*
 * Copyright 2025 cronn GmbH
 * SPDX-License-Identifier: AGPL-3.0-only
 */

package de.eshg.schoolentry.domain.model;

import de.eshg.domain.model.BaseEntityWithExternalId;
import de.eshg.lib.common.DataSensitivity;
import de.eshg.lib.common.SensitivityLevel;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.validation.constraints.NotNull;

@Entity
@DataSensitivity(value = SensitivityLevel.PROTECTED)
public class ProcedureLabel extends BaseEntityWithExternalId {

  @NotNull
  @Column(unique = true)
  private String name;

  private String description;

  @NotNull private String hexColor;

  private boolean readonly;

  public String getName() {
    return name;
  }

  public void setName(String name) {
    this.name = name;
  }

  public String getDescription() {
    return description;
  }

  public void setDescription(String description) {
    this.description = description;
  }

  public String getHexColor() {
    return hexColor;
  }

  public void setHexColor(String hexColor) {
    this.hexColor = hexColor;
  }

  public boolean isReadonly() {
    return readonly;
  }

  public void setReadonly(boolean readonly) {
    this.readonly = readonly;
  }
}

File: packages/lib-employee-portal/src/features/procedureLabels/api/models/ProcedureLabel.ts
/**
 * Copyright 2026 cronn GmbH
 * SPDX-License-Identifier: Apache-2.0
 */

import { BaseEntity, mapBaseEntity } from "../../../../api/models/BaseEntity";
import { Versioned, mapVersioned } from "../../../../api/models/Versioned";

export interface ProcedureLabel extends BaseEntity, Versioned {
  readonly name: string;
  readonly description?: string;
  readonly hexColor: string;
  readonly readonly?: boolean;
}

export interface ProcedureLabelResponse {
  id: string;
  version: number;
  name: string;
  description?: string;
  hexColor: string;
  readonly?: boolean;
}

export function mapProcedureLabel(
  response: ProcedureLabelResponse,
): ProcedureLabel {
  return {
    ...mapBaseEntity(response),
    ...mapVersioned(response),
    name: response.name,
    description: response.description,
    hexColor: response.hexColor,
    readonly: response.readonly,
  };
}

export function mapProcedureLabels(
  response: ProcedureLabelResponse[],
): ProcedureLabel[] {
  return response.map(mapProcedureLabel);
}
